File: Python/BOJ_1018_체스판 다시 칠하기.py
import sys

n, m = map(int, sys.stdin.readline().split())
chess = []
for i in range(n):
    chess.append(sys.stdin.readline().rstrip())

minimum = n * m
for j in range(n - 7):
    for k in range(m - 7):
        
        w_first = 0
        char = 'W'
        change = 0
        for a in range(j, j + 8):
            if change // 8 == 1:
                if char == 'W':
                    char = 'B'
                    change = 0
                else:
                    char = 'W'
                    change = 0
            for b in range(k, k + 8):
                if chess[a][b] != char:
                    w_first += 1
                if char == 'W':
                    char = 'B'
                elif char == 'B':
                    char = 'W'
                change += 1
        
        b_first = 0
        char = 'B'
        change = 0
        for c in range(j, j + 8):
            if change // 8 == 1:
                if char == 'B':
                    char = 'W'
                    change = 0
                else:
                    char = 'B'
                    change = 0
            for d in range(k, k + 8):
                if chess[c][d] != char:
                    b_first += 1
                if char == 'B':
                    char = 'W'
                elif char == 'W':
                    char = 'B'
                change += 1
                    
        if min(w_first, b_first) < minimum:
            minimum = min(w_first, b_first)

print(minimum)

File: C/BOJ_1018_체스판 다시 칠하기.c
#include <stdio.h>
#include <stdlib.h>

int min(int a, int b)
{
    if (a < b) return a;
    else return b;
}

int main()
{
    char** table;
    char* temp;
    char now;
    int column, row, w_first, b_first, change, minimum = 10e8;
    scanf("%d %d", &column, &row);

    temp = (char*)malloc(sizeof(char) * (row + 1));
    table = (char**)malloc(sizeof(char*) * column);
    for (int i = 0; i < column; i++){
        table[i] = (char*)malloc(sizeof(char) * row);
    }

    for (int i = 0; i < column; i++){
        scanf("%s", temp);
        for (int j = 0; j < row; j++){
            table[i][j] = temp[j];
        }
    }

    for (int i = 0; i < column - 7; i++){
        for (int j = 0; j < row - 7; j++){

            w_first = 0;
            now = 87; // W = 87, B = 66
            change = 0;
            for (int a = i; a < i + 8; a++){
                if (change / 8 == 1){
                    if (now == 87){
                        now = 66;
                        change = 0;
                    }
                    else{
                        now = 87;
                        change = 0;
                    }
                }
                for (int b = j; b < j + 8; b++){
                    if (table[a][b] != now) w_first++;
                    if (now == 87) now = 66;
                    else if (now == 66) now = 87;
                    change++;
                }
            }

            b_first = 0;
            now = 66;
            change = 0;
            for (int a = i; a < i + 8; a++){
                if (change / 8 == 1){
                    if (now == 66){
                        now = 87;
                        change = 0;
                    }
                    else{
                        now = 66;
                        change = 0;
                    }
                }
                for (int b = j; b < j + 8; b++){
                    if (table[a][b] != now) b_first++;
                    if (now == 66) now = 87;
                    else if (now == 87) now = 66;
                    change++;
                }
            }

            if (min(w_first, b_first) < minimum)
                minimum = min(w_first, b_first);
            
        }
    }

    printf("%d", minimum);
    free(temp);
    for (int i = 0; i < column; i++){
        free(table[i]);
    }
    free(table);
    return 0;
}